### PR TITLE
gsi_ncdiag surface pressure converter fixes

### PR DESCRIPTION
**Pull request by FanHan-NOAA**
## Description

This PR adds temporary fixes for some issues found with gsi ncdiag conversion of surface pressure observations. 

- add checks for processing GSI obs error and hofx, if the unit is hPa, convert to Pa
- check for large value (~1e9) of surface pressure observation, convert to missing value

### Issue(s) addressed

N/A 

## Acceptance Criteria (Definition of Done)

Test and approve by 2 reviewers.

## Dependencies

N/A

## Impact

N/A

### CODE REVIEW

File: src/gsi-ncdiag/gsi_ncdiag.py [hunk 1]
```diff
@@ -761,13 +761,17 @@ def toIODAobs(self, OutDir, clobber=True, platforms=None):
...
                     obserr = self.var('Errinv_Input')[idx]
                     mask = obserr < self.EPSILON
                     obserr[~mask] = 1.0 / obserr[~mask]
                     # below is a temporary hack until missing ObsError support returns to IODA/UFO
                     obserr[mask] = 1e8
                     # obserr[mask] = self.FLOAT_FILL
                     # obserr[obserr > 4e8] = self.FLOAT_FILL
+                    # convert surface_pressure error to Pa from hPa
+                    if v == 'ps' and np.nanmin(obserr) < 10:
+                        obserr = obserr * 100
```
Comment (CoryMartin-NOAA) [inline src/gsi-ncdiag/gsi_ncdiag.py:774]: ah, good catch.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Review by weihuang-jedi — APPROVED** [review]

**Comment by FanHan-NOAA** [comment]
@srherbener Thanks for checking the test failures. I just pushed a commit, hopefully the coding norm one will pass now.

**Review by srherbener — APPROVED** [review]
@FanHan-NOAA thanks for fixing the coding norms error.

**Comment by srherbener** [comment]
I just verified on my Mac that the tropomi test is failing (in the same manner) in the current develop branch so I will go ahead and merge this PR.

**Pull request merged**